### DISCUSSION TITLE
[@types/node] zlib.unzip should accept a Buffer per NodeJS documentation

### DISCUSSION THREAD

**Discussion by rdsedmundo** [DefinitelyTyped/DefinitelyTyped#55444] [Issues with a @types package] [answered]
https://nodejs.org/docs/latest-v14.x/api/zlib.html#zlib_zlib_unzip_buffer_options_callback

What we have typed is:
```
    type InputType = string | ArrayBuffer | NodeJS.ArrayBufferView;

    function unzip(buf: InputType, callback: CompressCallback): void;
```

[a comment by an automated account is not shown]

**Comment by SimonSchick** [comment, marked as answer]
This is already supported (and tested), did you test this use case before writing this bug?
`NodeJS.ArrayBufferView` is defined as such:

```
    type TypedArray =
        | Uint8Array
        | Uint8ClampedArray
        | Uint16Array
        | Uint32Array
        | Int8Array
        | Int16Array
        | Int32Array
        | BigUint64Array
        | BigInt64Array
        | Float32Array
        | Float64Array;
    type ArrayBufferView = TypedArray | DataView;
```

Buffer inherits from `Uint8Array`, therefor `unzip` (and others) do support `Buffer`.

Test cases:

https://github.com/SimonSchick/DefinitelyTyped/blob/be86e4888de65d5e7abf8ce4415b70bcdbe550e5/types/node/test/zlib.ts#L87-L88

**Reply by rdsedmundo** [reply]
Yes, I reported because I got an error, I think it might be related to promisify then, I'll have another look and get back to you.

**Reply by rdsedmundo** [reply]
You're right, it was because of a cast someone had done elsewhere in the code and I didn't notice, sorry.
